packages/dds/tree/src/simple-tree/proxies.ts

```diff
@@ -142,37 +143,44 @@ export function createObjectProxy<TSchema extends FlexObjectNodeSchema>(
 	// TODO: Although the target is an object literal, it's still worthwhile to try experimenting with
 	// a dispatch object to see if it improves performance.
 	const proxy = new Proxy(targetObject, {
-		get(target, key): unknown {
-			const field = getFlexNode(proxy).tryGetField(key as FieldKey);
+		get(target, viewKey): unknown {
+			const storedKey = getFlexObjectKey(viewKey as FieldKey, schema);
```

Comment:
Since this is in the field access code path, which is pretty perf sensitive, I think it might be worth taking a different approach here.

Instead of doing all the work each time the field is accessed, you can store a map from APIKey->FieldKey (ex: create it before declaring the proxy as a local and just use it in here and in set).

That might actually be simpler, and should be slightly faster.

Reply:
even better would be to create that map once (ex: lazily cached based on schema in the schema), then assign it to a local here and use that. We actually could just cache the whole second parameter to the proxy constructor based on schema and avoid caching the file map seperartly

Reply:
I will +1 that we have historically been pretty sensitive to adding cost (map lookups, etc.) to data reads. So if there's an optimization here that isn't too complicated, we should try it.

Reply:
Most recent changes include caching for the view key => stored key mapping (for object proxies, specifically, since they are the only proxy kinds for which this is relevant).